narwhals/_pandas_like/dataframe.py

```diff
@@ -335,6 +318,31 @@ def generate_unique_token(
                     ),
                 )
 
+        if how == "anti":
+            indicator_token = generate_unique_token(
+                n_bytes=8, columns=[*self.columns, *other.columns]
+            )  # pragma: no cover
+
+            other_ = (
+                other._dataframe.loc[:, right_on]
+                .rename(  # rename to avoid creating extra columns in join
+                    columns=dict(zip(right_on, left_on))  # type: ignore[arg-type]
+                )
+                .drop_duplicates(ignore_index=True)
+            )
+            return self._from_dataframe(
+                self._dataframe.merge(
+                    other_,
```

Comment:
why nnot just reassign to `other`?
```suggestion
            other = (
                other._dataframe.loc[:, right_on]
                .rename(  # rename to avoid creating extra columns in join
                    columns=dict(zip(right_on, left_on))  # type: ignore[arg-type]
                )
                .drop_duplicates(ignore_index=True)
            )
            return self._from_dataframe(
                self._dataframe.merge(
                    other,
```